### PR TITLE
add softmax_switch for softmax_with_cross_entropy_op, test=develop

### PR DESCRIPTION
**Pull request by chajchaj**
### PR types
 New features

### PR changes
OPs 

### Describe
add softmax_switch for softmax_with_cross_entropy_op
softmax_with_cross_entropy是先计算softmax，再计算cross_entropy。
为了满足用户单独使用cross_entropy的需求，本次升级增加了一个开关softmax_switch，
实现纯cross_entropy的功能。

### CODE REVIEW

**Comment by Superjomn** [comment]
我确认Op版本没问题，推理这边会关注这个Op在推理上的变化

最终看其他人的review意见吧

**Comment by zhiqiu** [comment]
LGTM for shareDataWith, but why not use cross_entropy directly?

**Comment by chajchaj** [comment]
> 
> 
> LGTM for shareDataWith, but why not use cross_entropy directly?
cross_entropy doesn't support "axis" attribute

**Review by zhangting2020 — APPROVED** [review]
LGTM. However, `skip_check_grad_ci` is not used in the code.

**Review by zhiqiu — APPROVED** [review]
LGTM for ShareDataWith

**Review by chalsliu — APPROVED** [review]
